File: exercise/php/day13/src/SantaMarket/Offer.php
<?php

namespace SantaMarket;

class Offer
{
    public Product $product;
    public SpecialOfferType $offerType;
    public float $argument;

    public function __construct(SpecialOfferType $offerType, Product $product, float $argument)
    {
        $this->offerType = $offerType;
        $this->product = $product;
        $this->argument = $argument;
    }
}

File: exercise/java/day13/src/main/java/santamarket/model/Offer.java
package santamarket.model;

public class Offer {
    private final Product product;
    SpecialOfferType offerType;
    double argument;

    public Offer(SpecialOfferType offerType, Product product, double argument) {
        this.offerType = offerType;
        this.argument = argument;
        this.product = product;
    }
}
